workspace/php2/src/XSolveSecurityBundle/Controller/UserController.php

```diff
@@ -19,17 +20,70 @@ public function RestLoginAction(Request $request) {
       $autorizationManager = $this->get('x_solve_security.xsolve_autorization_manager');
       $authenticatedToken = $autorizationManager->getToken($username, $password);
 
-      if ($authenticatedToken) {
-	 $this->getUser()->assignNewToken();
-	 $this->getDoctrine()->getManager()->flush();
+      if (!$authenticatedToken) {
+
+	 throw new AccessDeniedException('unable to find token');
+      }
+      $this->getUser()->assignNewToken();
+      $this->getDoctrine()->getManager()->flush();
+
+      return (new JsonResponse())->setData(
+		      ['token' => $this->getUser()->getToken()->getToken()]);
+   }
+   public function showUserAction($id) {
+
+      $user = $this->getDoctrine()
+	      ->getRepository('XSolveSecurityBundle:User')
+	      ->find($id);
+
+      if (!$user) {
+
+	 throw $this->createNotFoundException(
+		 'No found user');
+      }
+
+      return (new JsonResponse())->setData(
+		      ['login' => $user->getUsername(),
+			'password' => $user->getPassword()]);
+   }
+   public function showAllUsersAction() {
+
+      $users = $this->getDoctrine()
+	      ->getRepository('XSolveSecurityBundle:User')
+	      ->findAll();
+
+      $tab = []; //tab contain id, usernames and passwords of all user in database
 
+      foreach ($users as &$user) {
 
-	 return (new JsonResponse())->setData(
-			 ['login' => $username,
-			  'password' => $username,
-		          'token' => $this->getUser()->getToken()->getToken()]);
+	 $element = ['id' => $user->getId(),
+	     'username' => $user->getUsername(), ];
+	 array_push($tab, $element);
       }
-      throw new AccessDeniedException('unable to find token');
+
+
+      $response = new jsonResponse();
+      $response->setData($tab);
+
+      return $response;
+   }
+
+   public function registerAction(Request $request) {
+      
+      $username = $request->request->get('login');
+      $password = $request->request->get('haslo');
+      $user = new User();

```

Comment:
Tutaj informacja o moim stylu i wcale nie musisz jej brać do siebie, ja osobiście lubię sobie deklarować zmienne na początku a potem z nich korzystać, daje to informacje od razu, z czego będzie można w danej metodzie korzystać, więc ja bym to napisał w ten sposób

```
$user = new User(); // najpierw zmienne które są tworzony

$username = $request->request->get('login'); // potem te które można skądś przypisać 
$password = $request->request->get('haslo');

$user->setUsername($username); // a potem operacje na nich
$user->setPassword($password);